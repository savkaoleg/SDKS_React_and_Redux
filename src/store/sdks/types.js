export const SET_SDKS = 'SET_SDKS'
