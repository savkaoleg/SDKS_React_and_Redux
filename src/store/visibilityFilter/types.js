export const HANDLE_CHANGE = 'HANDLE_CHANGE'
export const HANDLE_SEARCH_CHANGE = 'HANDLE_SEARCH_CHANGE'
