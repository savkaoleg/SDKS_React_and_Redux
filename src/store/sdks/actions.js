import { SET_SDKS } from './types'

export function setSdks(payload) {
  return {
    type: SET_SDKS,
    payload
  }
}
