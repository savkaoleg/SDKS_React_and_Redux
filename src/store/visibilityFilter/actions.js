import { HANDLE_CHANGE, HANDLE_SEARCH_CHANGE } from './types'

export function handleChange(newType, newAditional) {
  return {
    type: HANDLE_CHANGE,
    newType,
    newAditional
  }
}

export function handleSearcChange(newType, searchQuery) {
  return {
    type: HANDLE_SEARCH_CHANGE,
    newType,
    searchQuery
  }
}
