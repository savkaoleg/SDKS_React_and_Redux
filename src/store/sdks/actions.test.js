import expect from 'expect'
import { setSdks } from './actions'
import { SET_SDKS } from './types'

const data = [
  {
    title: 'Accengage',
    id: 'accengage',
    tags: ['marketing-automation']
  },
  {
    title: 'AdBuddiz',
    id: 'adbuddiz',
    tags: ['ad-network']
  }
]

describe('>>> sdks > actions', () => {
  it('setSdks snapshot', () => {
    const setSdksAction = setSdks(data)
    expect(setSdksAction).toMatchSnapshot()
  })

  it('setSdks', () => {
    const setSdksAction = setSdks(data)
    expect(setSdksAction).toEqual({ type: SET_SDKS, payload: data })
  })
})
