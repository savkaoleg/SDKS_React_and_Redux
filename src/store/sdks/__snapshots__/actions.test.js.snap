// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> sdks > actions setSdks snapshot 1`] = `
Object {
  "payload": Array [
    Object {
      "id": "accengage",
      "tags": Array [
        "marketing-automation",
      ],
      "title": "Accengage",
    },
    Object {
      "id": "adbuddiz",
      "tags": Array [
        "ad-network",
      ],
      "title": "AdBuddiz",
    },
  ],
  "type": "SET_SDKS",
}
`;
